pandas/core/frame.py

```diff
@@ -6741,7 +6747,23 @@ def corr(self, method='pearson', min_periods=1):
                              "'spearman', or 'kendall', '{method}' "
                              "was supplied".format(method=method))
 
-        return self._constructor(correl, index=idx, columns=cols)
+        corr_mat = self._constructor(correl, index=idx, columns=cols)
+
+        if tri is not None:
+            mask = np.tril(np.ones_like(correl,
+                                        dtype=np.bool),
+                           k=-1)
+
+            if tri == 'lower':
+                return corr_mat.where(mask)
```

Comment:
Not the speed is probably important, but `corr+np.tril(corr*np.nan)` is probably faster than indexing.